pmd-core/src/main/java/net/sourceforge/pmd/ant/Formatter.java

```diff
@@ -63,18 +67,17 @@ public Renderer getRenderer() {
     public void start(String baseDir) {
 
         Properties properties = createProperties();
-        
+
         Charset charset;
         {
             String s = (String) properties.get("encoding");
             if (null == s) {
 
                 if (toConsole) {
-                    if (null == System.console()) {
-                        // pipe or redirect, no interactive console.
+                    s = getConsoleEncoding();
+                    if (null == s) {
```

Comment:
@pyxide doesn't this mean that, even if the output is piped (`System.console() == null`), we would take the value returned by `Console.encoding()`?

I would just used the reflection approach wen there is a console attached.